main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -74,10 +74,7 @@ object Kinder {
 
     val defs = visitDefs(root, taenv, oldRoot, changeSet)
 
-    // Should `instances` not be root.instances.map(ParOps.parMap)?
-    // I.e., for every trait, we check all instances in parallel so we get a parallel task for every instance
-    // instead of check all traits in parallel which spawns k threads each does synchronous code
-    val instances = ParOps.parMapValues(root.instances)(_.map(visitInstance(_, taenv, root)))
+    val instances = ParOps.parMapValues(root.instances)(ParOps.parMap(_)(visitInstance(_, taenv, root)))
```

Comment:
Oooh. I misunderstood. Don't do _nested parallelism_ that is not good. 

I thought you wanted to move the level, not do nested parallelism.

Reply:
Alright :)

Reply:
That was actually my initial idea but then I realized I could do this :D